x/precisebank/types/fractional_balance.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	fmt "fmt"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// FractionalBalance contains __only__ the fractional balance of an address.
+// We want to extend the current KAVA decimal digits from 6 to 18, thus 12 more
+// digits are added to the fractional balance. With 12 digits, the maximum
+// value of the fractional balance is 1_000_000_000_000 - 1.
+// We subtract 1, as 1 more will roll over to the integer balance.
+var MAX_FRACTIONAL_AMOUNT = sdkmath.NewInt(1_000_000_000_000).SubRaw(1)
```

Comment:
Since this can't be a `const`, I'd lean towards making this private within the module and offer a public function outside consumers (if any).

Reply:
Though I also assume at some point, this max will be based on the default conversion factor, but the above should still apply.

Reply:
i didn't see this comment. i'd prefer it to be an exported value rather than a getter method for an underlying value. IMO it's more idiomatic

though it should be named `MaxFractionalAmount`, and not all caps.

my two cents anyway

Reply:
hmm yeah that does seem cleaner, or a 3rd option being like the same as `sdkmath.ZeroInt()`?

```go
func MaxFractionalInt() sdkmath.Int { return sdkmath.NewInt(1_000_000_000_000).SubRaw(1) }
```

Some overhead of making a new one each time its called though, not entirely sure why they do that since Ints should be immutable with via public methods

Reply:
ah wait I guess not using a global var prevents it being re-assigned a new value.

var MaxFractionalInt = ...

makes it possible to do `MaxFractionalInt = sdk.NewInt(xx)` when the func should be easier to enforce it isn't mutable